libs/nio/src/main/java/org/opensearch/nio/ServerChannelContext.java

```diff
@@ -124,7 +129,10 @@ protected static SocketChannel accept(ServerSocketChannel serverSocketChannel) t
         try {
             assert serverSocketChannel.isBlocking() == false;
             SocketChannel channel = AccessController.doPrivileged((PrivilegedExceptionAction<SocketChannel>) serverSocketChannel::accept);
-            assert serverSocketChannel.isBlocking() == false;
+            if (serverSocketChannel.isBlocking() == true) {
+                channel.close();
+                throw new AssertionError("serverSocketChannel is blocking.");
```

Comment:
Note: I want to keep a similar behavior with the previous "assert" statement, so threw this AssertionError().